conans/client/generators/compiler_args.py

```diff
@@ -20,30 +20,35 @@ def filename(self):
     def compiler(self):
         return self.conanfile.settings.get_safe("compiler")
 
+    @property
+    def _settings(self):
+        settings = self.conanfile.settings.copy()
```

Comment:
I need this `hack` only to make strange GCC generator to work, as it uses flags assuming compiler is GCC even for Visual Studio profile, so it might get very strange mix of flags

Reply:
I don't get the change. The ``gcc`` generator is deprecated in favor of this ``compiler_args`` generator. Did yo mean gcc compiler?

Reply:
it is deprecated. that's right, no doubts here.
however, it's still tested in some unit tests. I had to change interface of some internal helpers which return compiler flags. unfortunately, GCC generator used to rely on these helpers, so I needed to adapt this one as well, so tests aren't broken.
previously, such helpers received individual settings values, e.g.:
```
def architecture_flag(compiler, arch, os=None):
```
now we just receive the whole settings object (in order to handle `compiler.base` values, if they exist)
```
def architecture_flag(settings):
```

the GCC generator used to rely on mixed settings - all values were same, except compiler, which was forced to be GCC. so, e.g. if you had `compiler=Visual Studio` in your profile, GCC generator computed flags for `compiler=gcc`, but the rest of settings were taken from the Visual Studio (e.g. `compiler.runtime` and so on). this resulted in weird mixture of flags for some compilers (most notably, Visual Studio).

as we can't break, I think I have to preserve the existing behavior, it means GCC generator still must produce the strange mixture of Visual Studio + GCC flags, otherwise we will consider it as a bug (breaking change).

however, as now helpers accept the whole `settings` object, it's getting complicated - I can't just assign `settings.compiler` to the passed `settings` object - I think generator shouldn't modify objects passed to it, treating them as a read only.

the one alternative is to make a temporary copy of the `settings` object and modify the `settings.compiler` value on the copied instance, so the original object is untouched. that was the easiest way for me.

another way, is to still provide second facade for the helpers - overload functions which will accept individual settings values. IMO this is a way too complicated change just to preserve the behavior of the GCC generator.

in any case, I suppose this is not a big deal, as (I hope) too few people are using this GCC generator.

Reply:
Yes, I understand it but as said, we don't even have documentation for that gcc generator https://docs.conan.io/en/latest/reference/generators/gcc.html so maybe it can be removed. WDYT @memsharded?

Reply:
if you're okay to remove it, it's easy. I just worry if it might be breaking, even for minority of users, if someone, for instance, is using GCC generator in his CI scripts or somewhere else.

Reply:
Is deprecated, but not removed, it is 1.X "stable", cannot be removed atm.